internal/xcrd/crd_test.go

```diff
@@ -1164,16 +1152,62 @@ func TestForCompositeResourceClaim(t *testing.T) {
 					},
 				},
 			},
-		},
+		}
 	}
 
-	got, err := ForCompositeResourceClaim(d)
-	if err != nil {
-		t.Fatalf("ForCompositeResourceClaim(...): %s", err)
+	type test struct {
+		name   string
+		schema string
+		want   *extv1.CustomResourceDefinition
+	}
+	tests := []test{
+		{name: "standard case", schema: fmt.Sprintf(schemaFmt, ""), want: want(63)},
+		{name: "small name restriction", schema: fmt.Sprintf(schemaFmt, `"metadata":{"properties":{"name":{"type":"string","maxLength":10}}},`), want: want(10)},
+		{name: "too large name restrictions", schema: fmt.Sprintf(schemaFmt, `"metadata":{"properties":{"name":{"type":"string","maxLength":100}}},`), want: want(63)},
```

Comment:
nit: took me a bit to understand the `want: want(63)` part, maybe a more explicit name for the `want` function could be helpful, also because we usually have a `want` struct defined to wrap the wanted outcome.

Reply:
done